### ISSUE TITLE
Separate re-scalable Instrument GUI

### ISSUE THREAD

**Issue by TheScienceNerd** [StavWasPlayZ/Genshin-Instruments#8]
So the mod is finally working with the new update, and when I go to open it my whole screen is filled with the Instrument GUI since I have my GUI scale to the max so I can really see everything, but I wasn't expecting the Instrument GUI to also get re-scaled.
![image](https://github.com/StavWasPlayZ/Genshin-Instruments/assets/67557526/f6b4f06f-1eac-4dd5-b648-0584bf14de1a)

Is it possible to get an option to rescale it so even if my GUI scale setting is at 5x, I can dial back the scale to what it would normally look like at a 2x GUI scale?

**Comment by StavWasPlayZ** [comment]
Hey there, I patched an update to fix GUI scale 4 in version v3.2.2. Please see if it helps: https://www.curseforge.com/minecraft/mc-mods/genshin-instruments/files

By the way, I assume this is Fabric you are talking about here. Please report Fabric issues to the [Fabric repository](https://github.com/StavWasPlayZ/Genshin-Instruments-Fabric/issues) in the future.

Thanks!

**Comment by StavWasPlayZ** [comment]
I will make an adjustment nevertheless to also support 5x scales. I personally do not have any monitor that can support it, so I'll have to send you directly a test version for it. Could you please send me your Minecraft version so I can provide you with the JAR?

**Comment by StavWasPlayZ** [comment]
Fabric: https://github.com/StavWasPlayZ/Genshin-Instruments-Fabric/releases/tag/1.20.1-3.2.2.1  
Forge: https://github.com/StavWasPlayZ/Genshin-Instruments/releases/tag/3.2.2.1-1.20.1

Note that v3.2.2.1 is a test version made for this problem in specific. If you declare this problem solved, I'll release the full version by tomorrow or so. Looking forward to your response,

---

Making a custom GUI scale option is not the easiest for the sole reason that the options screen is already flooded with options. You also cannot see the options screen when the notes are this huge, anyways...

Regardless, I have made the default scaling size of note buttons be `40`, whereas previously it was `GUI scale * 18`; essentially 90 for you. Which is rather insane.

Please tell me if you'd like further reduction from the size.

**Comment by TheScienceNerd** [comment]
So I downloaded the 3.2.2.1 version and installed it but it seems to have done nothing. At 5x GUI scale it still looks like the original picture I sent and is still affected by the GUI scaling.

**Comment by StavWasPlayZ** [comment]
I see. In this case, I have made further reductions to 5x scales:

Forge: https://github.com/StavWasPlayZ/Genshin-Instruments/releases/tag/3.2.3-1.20.1  
Fabric: https://github.com/StavWasPlayZ/Genshin-Instruments-Fabric/releases/tag/1.20.1-3.2.3

Instead of size `40`, it is now `20`. If you find this still is too huge/small, please instruct me to change the size accordingly.
Sorry to bother you in this matter, I simply have no way to set my scalings this high to test it myself...

Thanks!

**Comment by StavWasPlayZ** [comment]
TL;DR: v3.2.3 should fix the problem. Please re-open the issue if you find this still problematic, but I doubt I could do anything further :(

Also, avoid using Auto scale. Use a fixed one, i.e `5x`.

---

I have settled on the value `35`, as I have successfully made a VM to support high resolutions. This is the result I got:
![image](https://github.com/StavWasPlayZ/Genshin-Instruments/assets/89315181/06e19804-6d77-455b-90c7-b2d43f27d2ca)

I also tried to support 6x scalings (to the best of my abilities):
![image](https://github.com/StavWasPlayZ/Genshin-Instruments/assets/89315181/a1028392-c81e-4523-89ce-e55f6d086565)

However, auto scalings act up very weird. I default-ed them to be on `35`, as it works best on scales 1-3:

1. ![image](https://github.com/StavWasPlayZ/Genshin-Instruments/assets/89315181/1041adce-0343-4905-8b3b-3c90a662f427)
2. ![image](https://github.com/StavWasPlayZ/Genshin-Instruments/assets/89315181/3cddc863-9238-45c4-81d0-5843cb80808c)
3. ![image](https://github.com/StavWasPlayZ/Genshin-Instruments/assets/89315181/f620ff65-d803-4760-80ec-9e801a20a975)
But 6 is just a black sheep: ![image](https://github.com/StavWasPlayZ/Genshin-Instruments/assets/89315181/55b26e66-cd10-4181-8d97-5ab1d019188a)

So I would recommend against using auto scales with this mod, especially with grid instruments. Sorry for the inconvenience, but I have little I can do to minimize the effects of this. I don't know any method to directly tell Minecraft the size of the buttons to render, it just kinda decides it on its own once you set up the GUI scale. All I can do is just dynamically reduce/increase each case of scale to a guessed size I see fit:

```java
public int getNoteSize() {
    return switch (Minecraft.getInstance().options.guiScale().get()) {
        case 1 -> 36;
        case 2 -> 46;
        case 3 -> 48;
        case 4 -> 40;
        case 5 -> 35;
        case 6 -> 30;

        default -> 35;
    };
}
```

I hope this patch will, at least, minimize the issue. 🥹
